script/download-assets.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-console */
+
+require('isomorphic-fetch');
+const buckets = require('./constants/buckets');
+
+function downloadAssets(buildOptions) {
+  return async (files, smith, done) => {
+    const bucket = buckets[buildOptions.buildtype];
+    const fileManifestPath = 'generated/file-manifest.json';
+
+    console.log(`Downloading assets from ${bucket}...`);
+
+    const fileManifestRequest = await fetch(`${bucket}/${fileManifestPath}`);
+    const fileManifest = await fileManifestRequest.json();
+
+    files[fileManifestPath] = {
+      path: fileManifestPath,
+      contents: new Buffer(JSON.stringify(fileManifest)),
+    };
+
+    const entryNames = Object.keys(fileManifest);
```

Comment:
There should be another filter here to account for the `entry` build arg, but it got complicated also trying to account for fonts, maps, css, etc. Plenty of room for further optimization.